src/GenerateAnalyzerRulesets/Program.cs

```diff
@@ -234,7 +246,80 @@ string getRuleActionCore(bool enable)
                 }
             }
 
-            return 0;
+            void createPropsFile()
+            {
+                if (string.IsNullOrEmpty(propsFileDir) || string.IsNullOrEmpty(propsFileName))
+                {
+                    Debug.Assert(!containsPortedFxCopRules);
+                    return;
+                }
+
+                var fileContents =
+$@"<Project DefaultTargets=""Build"" xmlns=""http://schemas.microsoft.com/developer/msbuild/2003"">
+  {getCodeAnalysisTreatWarningsNotAsErrors()}{getRulesetOverrides()}
+</Project>";
+                var directory = Directory.CreateDirectory(propsFileDir);
+                var fileWithPath = Path.Combine(directory.FullName, propsFileName);
+                File.WriteAllText(fileWithPath, fileContents);
+            }
+
+            string getCodeAnalysisTreatWarningsNotAsErrors()
+            {
+                var allRuleIds = string.Join(';', allRulesByAssembly.Values.SelectMany(l => l.Keys).Distinct());
+                return $@"
+  <!-- 
+    This property group prevents the rule ids implemented in this package to be bumped to errors when
+    the 'CodeAnalysisTreatWarningsAsErrors' = 'false'.
+  -->
+  <PropertyGroup Condition=""'$(CodeAnalysisTreatWarningsAsErrors)' == 'false'"">
+    <WarningsNotAsErrors>$(WarningsNotAsErrors);{allRuleIds}</WarningsNotAsErrors>
+  </PropertyGroup>";
+            }
+
+            string getRulesetOverrides()
+            {
+                if (containsPortedFxCopRules)
+                {
+                    var rulesetOverridesBuilder = new StringBuilder();
+                    foreach (var category in categories.OrderBy(k => k))
+                    {
+                        // Each rule entry format is: -[Category]#[ID];
+                        // For example, -Microsoft.Design#CA1001;
+                        var categoryPrefix = $"      -Microsoft.{category}#";
+                        var entries = allRulesByAssembly.Values
+                                          .SelectMany(l => l)
+                                          .Where(ruleIdAndDescriptor => ruleIdAndDescriptor.Value.Category == category &&
+                                                                        FxCopWellKnownDiagnosticTags.IsPortedFxCopRule(ruleIdAndDescriptor.Value))
+                                          .OrderBy(ruleIdAndDescriptor => ruleIdAndDescriptor.Key)
+                                          .Select(ruleIdAndDescriptor => $"{categoryPrefix}{ruleIdAndDescriptor.Key};")
```

Comment:
This is disable rules in FxCopCmd.exe?  For the security rules, I know there are some where the FxCop and Roslyn analyzer IDs aren't a 1:1 mapping.

Reply:
Yes, it does. I verified running FxCopCmd.exe with the generated props file that included every rule was causing CA1061 in FxCopCmd.exe (basically unknown rule diagnostic). I have only retained the publically shipping rules as PortedFxCopRules, otherwise CA1061 shows up when RunCodeAnalysis is executed in VS.

Reply:
Oh okay, I think this is fine for the publicly available ones.